grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -679,12 +679,16 @@ class OrderService {
                                               supplier: supplier,
                                               sourceName: sourceName]
                         ProductSupplier productSupplier = productSupplierDataService.getOrCreateNew(supplierParams, false)
+                        def supplierParamFilled = supplierCode || manufacturerName || manufacturerCode
```

Comment:
Let's come up with a better variable name here. What is the gist of this boolean value? The required fields for supplier are populated? Or something else?

Reply:
yep, I also don't like this name, but I didn't have any other idea.
The gist is to determine, if a user filled any of those three fields, as by them we can search for a product source.
I had to do it, because there is a rare case where it could match a product source with all those fields empty, but what if a user didn't want to assign any product source (he left those three empty) and it matched a product source with all those three fields empty and it's an inactive product source? User would be punished for that, that this product source is inactive, but the user might think: "_What? I didn't even want to assign a product source, I left the fields empty, so what did it match_?" 
That's the reason I "mix the soup" below

Reply:
my only idea for name was something like this: `searchTermFieldsForSupplierFilled` but it was way too long and did not come up with anything else, but maybe this could help you with helping me to provide a better name? :laughing: